### ISSUE TITLE
Cannot bring GE to front

### ISSUE THREAD

**Issue by gerhardol** [gitextensions/gitextensions#9752]
### Environment

- Git Extensions 33.33.33
- Build 4240f4903d3b0d7ec70e73dfc4c5425023f8ad7e
- Git 2.34.0.windows.1
- Microsoft Windows NT 10.0.19043.0
- .NET 5.0.12
- DPI 96dpi (no scaling)

### Issue description

In some situations (normally at work) GE does not respond to being activated. Alt-Tab, clicking in the taskbar has no effect.
Show desktop and other tricks does not help to show the app.
It is possible to close the application from the taskbar. (So it does not seem to be an open dialog.)

### Steps to reproduce

Unknown, occurs mostly at work after some hour usage.

### Did this work in previous version of GitExtensions?

This did not occur prior to .NET5 migration, but it probably started some time after.

**Comment by vbjay** [comment]
Usually caused by owned fork maybe not visible but is there. So it focused
owned form instead but you just can't see it.

**Comment by gerhardol** [comment]
> Usually caused by owned fork maybe not visible but is there. So it focused owned form instead but you just can't see it.

Sure, but what change in master is causing thisto happen more often?
If it occurs to others, it is a blocker for 4.0
(How many are using master for daily work now?)

**Comment by vbjay** [comment]
> > Usually caused by owned fork maybe not visible but is there. So it focused owned form instead but you just can't see it.
> 
> Sure, but what change in master is causing thisto happen more often?
> If it occurs to others, it is a blocker for 4.0
> (How many are using master for daily work now?)

Do you have multiple monitors different scaling?  Might be related to that. Frequency. Master has moved to 6.0 or is still at 5.0 but either way it's not framework any more.  I'm betting dpi handling is a lot different.

 You would think clicking or alt tabbing into a parent window that has an owned form would make sure owned form was not just brought to focus but visible and not lost in out of view area by the framework.

**Comment by RussKie** [comment]
FWIW I'm observing that when the app is minimised the first click on the
task bar icon doesn't restore the app, instead I hear a beep sound. On the
2nd click the app restores... This is happening to the v3.x and v4.x
versions though for me.

**Comment by gerhardol** [comment]
The miniature on the taskbar is OK if there are several instances (it always is).
Sometimes I see the app when clicking the taskbar, it disappears immediately.

Sometimes the problem occurs by just clicking on the taskbar a couple of times.

Found a workaround:
I can restore the app by selecting on the taskbar, then Win-Up.

> FWIW I'm observing that when the app is minimized the first click on the
task bar icon doesn't restore the app, instead I hear a beep sound. On the
2nd click the app restores... This is happening to the v3.x and v4.x
versions though for me.

I have seen this too, but only after .NET5 what I recall.

**Comment by gerhardol** [comment]
The workaround with more than one instance and the icons are combined is to right-click and select restore.
This is super annoying.

**Comment by RussKie** [comment]
Have you observed this issue with the latest master?

**Comment by gerhardol** [comment]
> Have you observed this issue with the latest master?

Yes

**Comment by gerhardol** [comment]
When this occurs, the status in the toolbar is gray, like it is for the Dashboard below. (Not a problem here, but I cannot reproduce at home.)

![image](https://user-images.githubusercontent.com/6248932/166093707-6d27f185-3d03-4f85-9518-0c59eefe3f29.png)

You can use Restore also if only one instance is open

![image](https://user-images.githubusercontent.com/6248932/166093789-e38d941b-6df9-4fc3-8d5f-d3cde23e4dd0.png)

Occasionally, it seem to occur just activating another application, no way to reliably reproduce at all.

**Comment by asmwarrior** [comment]
I see the same issue here on my PC.

**Comment by gerhardol** [comment]
I can consistently reproduce with the following procedure:
* Start two GE instances with Git repos (not just dashboard).
* Minimize them (only one needed, but easier to see the problem).
* Close one of the instances while minimized
* Further opened instances (including started in the debugger) cannot be restored after minimized. (Dashboard can be restored)
* One instance can often be opened, but when you open a second the problem comes back

I used to get rid of the problem by exiting GE restored.
When writing this description, I am not able to get rid of the issue, also when deleting AppData/Roaming and AppData/Local

I eventually found some hooks in the debugger, submitting a PR for review
